src/node/db/PadManager.js

```diff
@@ -79,27 +80,25 @@ let padList = {
     await this.load();
 
     if (!this.sorted) {
-      this.list.sort();
+      this.cachedList = Array.from(this.list).sort();
       this.sorted = true;
     }
 
-    return this.list;
+    return this.cachedList;
   },
   addPad: function(name) {
     if (!this.initiated) return;
 
-    if (this.list.indexOf(name) == -1) {
-      this.list.push(name);
+    if (!this.list.has(name)) {
+      this.list.add(name);
       this.sorted = false;
     }
```

Comment:
We want to know if `add()` has actually inserted an element, because in that case we need assign `sorted = false` and invalidate `cachedList`. That's ok.

Proposal: instead of using `list.has()` (at cost O(log(n)) now that `list` is actually a `Set`), couldn't we always perform the `add()`, and compare the set size before and after the insertion? The cost of `Set.size()` is probably O(1), and in the case of successful insertion we would be doing half the work.

Reply:
`.has` and `.add` are `O(1)` because a `Set` is a hashtable

Reply:
Thanks, perfect!